mindsdb/integrations/handlers/ckan_handler/ckan_handler.py

```diff
@@ -1,101 +1,282 @@
-            ckan = rc(url)
+            self.connection = RemoteCKAN(url, apikey=api_key)
             self.is_connected = True
-            self.ckan = ckan
+            logger.info(f"Successfully connected to CKAN at {url}")
         except Exception as e:
-            return HandlerStatusResponse(False, f'Failed to connect to CKAN: {e}')
-        self.connection = ckan
-        return HandlerStatusResponse(True)
+            logger.error(f"Error connecting to CKAN: {e}")
+            raise ConnectionError(f"Failed to connect to CKAN: {e}")
 
-    def disconnect(self):
-        self.is_connected = False
+        return self.connection
 
     def check_connection(self) -> HandlerStatusResponse:
-        response = HandlerStatusResponse(False)
         try:
             self.connect()
-            result = self.ckan.action.status_show()
-            if 'datastore' not in result.get('extensions'):
-                """
-                If the CKAN instance does not have the datastore extension,
-                we can't use it.
-                """
-                response.message = 'CKAN datastore is not enabled'
-                response.status = False
-                self.is_connected = False
-                return response
-
+            return HandlerStatusResponse(success=True)
         except Exception as e:
-            logger.error(f'Error connecting to CKAN: {e}!')
-            self.is_connected = False
-            response.error_message = e
+            logger.error(f"Error checking connection: {e}")
+            return HandlerStatusResponse(success=False, error_message=str(e))
 
-        if response.success is False and self.is_connected is True:
-            self.is_connected = False
+    def call_ckan_api(self, method_name: str, params: dict):
+        connection = self.connect()
+        method = getattr(connection.action, method_name)
 
-        response.success = True
-        return response
+        try:
+            result = method(**params)
+            return result
+        except Exception as e:
+            logger.error(f"Error calling CKAN API: {e}")
+            raise RuntimeError(f"Failed to call CKAN API: {e}")
 
-    def query(self, query: ASTNode) -> HandlerResponse:
-        if not self.ckan:
-            self.connect()
-        query_str = self.renderer.get_string(query, with_failback=True)
-        return self.native_query(query_str)
-
-    def native_query(self, query_str: str) -> HandlerResponse:
-        """
-        Execute a query on the CKAN instance.
-        :param query_str: The query to execute.
-        :return: The response of the query.
-        """
-        if not self.ckan:
-            self.connect()
-        result = self.ckan.action.datastore_search_sql(sql=query_str)
-        if len(result.get('records')) > 0:
-            df = pd.DataFrame(result['records'])
-            response = HandlerResponse(RESPONSE_TYPE.TABLE, df)
+    def query(self, query: ast.Select) -> HandlerResponse:
```

Comment:
This method implementation is not required. The `APIHandler` interface already handles this. Let's remove it.